packages/vite/src/index.ts

```diff
@@ -129,7 +130,11 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
         if (viteConfig.command === 'serve') {
           const id = await generateImageID(srcURL, config, img)
           generatedImages.set(id, image)
-          metadata.src = path.posix.join(viteConfig?.server?.origin ?? '', basePath) + id
+        }
+        if (directives.has('inline')) {
+          metadata.src = `data:image/${metadata.format};base64,${(await image.toBuffer()).toString('base64')}`
+        } else if (viteConfig.command === 'serve') {
+          metadata.src = join(viteConfig?.server?.origin ?? '', basePath) + id
```

Comment:
this is going to use the `id` passed into `load` rather than the image ID as it was before. we should probably rename this to `imageId` to avoid having two variables with the same name

You need to remove the `const` from line 131, make that `imageId`, and add a `let imageId;` right before the `if (viteConfig.command === 'serve') {` check. GitHub won't let me make a suggestion on those lines since they're unchanged

Reply:
Updated